### DISCUSSION TITLE
How to add an attached file on top of allure report.

### DISCUSSION THREAD

**Discussion by everytimeloosinggame** [allure-framework/allure2#2721] [Questions & Support]
Hello everybody,

I want to add an existing attached file, generated from robot framework on top of the allure report. Currently I am calling the "Attach File" function from AllureLibrary  inside a Key Word in my robot script, as a result the file is being attached somewhere in between the report which is bit hard to navigate for the user.

I have attached two files called "current_file_attachement_location" which is the current implementation and "expected_file_attachment_location" which is where I want the location of the file to be. This will give you a better understanding of my problem.

I know I can just change the location of "Attach File" function call inside the Test Case section which will kind of solve this issue, but I don't want to do this as it makes the code bit complex. I am looking for alternative methods which can solve this issue.

Any suggestions or solutions are much appreciated :).

Thank you in advance.

**Comment by lubomsky** [comment]
Bump :)
I want to add attachement on `@AfterEach` (junit5), but files added that way are hidden in Tear down and under method, so they are easy to miss and hard to get.

I mean is there any simpler solution than implementing `AfterTestExecutionCallback`.

**Reply by gt-siberian153** [reply]
I've had the same issue. I've solved it by passing arguments with Return clause until it reach an acceptable level of deepness in the tree before logging the message.

Maybe it wont't be working for you with Junit. Im using Robot-Framework.

It' also possible to log your message with attachment with a failure status (with a try catch for example) 

```
Run Keyword And Ignore Error    Fail    msg=${videoIframesList}
```

Then you can write a bookmarklet JS function that open the report three of all nods with a failure status. This can be generated easily with a LLM and a js to bookmarklet converter that inline your JS then you store it in a browser bookmark and when your're browseing the report of the test case, you clic the bookmark and the tree opens and display all failures. I'm wondering why this is not made natively by allure-report ;-)

Btw, don't use attach function to avoid report html attachments to have a huge file size. I store them on an S3 bucket and generate relative URLS during the test execution and log them as html markup which keep report size quiet fit.
